### DISCUSSION TITLE
How to step through getItemOffersBatch() response array in PHP?

### DISCUSSION THREAD

**Discussion by JeffMash** [amzn/selling-partner-api-models#3485] [General] [answered]
Just wondering if there's a better way to access the values in the response for getItemOffersBatch() than a ton of foreach loops!

Right now, my code is something similar to this:

$ItemOffersBatchResponse = $ProductPricingApiInstance->getItemOffersBatch($body);

This is what it returns: https://github.com/jlevers/selling-partner-api/blob/main/docs/Model/ProductPricingV0/GetItemOffersBatchResponse.md

I want to step into the response and get a value for: response->body->payload->offers->is_buy_box_winner

If you can show me how to access THAT variable in PHP, please let me know!  Here is a sample of my code, but I don't know if it's the most efficient way to access the variables.

```
foreach($ItemOffersBatchResponse['responses'] as $Response)
	{
		foreach($Response['body'] as $ResponseBody)
		{
			foreach($ResponseBody as $key=>$value)
			{
				if($key == "asin")
					echo "Working on ASIN: " . $value, PHP_EOL;
				if($key == "offers")
				{
					// get Offer details
				}
			}
		}
	}
```

**Comment by JeffMash** [comment, marked as answer]
Figured it out.   My problem was not differentiating between arrays and objects.

**Reply by Barryw1969** [reply]
Hi

Is it possible you can show me a sample of the working solution.

I am quite new to sp api and PHP - I can get the data for getItemOffersBatch  I just cant seem to parser it.

I have tried everything - even your example above.  I know I am getting the data because when I var_dump from response it shows everything.

**Comment by JeffMash** [comment]
> Hi
> 
> Is it possible you can show me a sample of the working solution.
> 
> I am quite new to sp api and PHP - I can get the data for getItemOffersBatch I just cant seem to parser it.
> 
> I have tried everything - even your example above. I know I am getting the data because when I var_dump from response it shows everything.

In my original posting, I was trying to use the function getItemOffersBatch(), but it looks like I was able to accomplish what I was looking for by using this other function: getListingOffersBatch()

(Item vs Listing)

I use the Listing function to access (or list) all the offers on my products (so that I can grab the information I need, like seeing who has the Buy Box, or seeing how much the current listing price is, etc).

Here are the important pieces of my code, in case you can use it as a reference to adapt to your own situation:

[code]
$arrRequests[] = new ProductPricing\ListingOffersRequest([
					'uri' => '/products/pricing/v0/listings/**item_sku_goes_here**/offers',
					'method' => 'GET',
					'item_condition' => 'New',
					'marketplace_id' => 'Your_Marketplace_ID',
				]);
$body = new ProductPricing\GetListingOffersBatchRequest([
			'requests' => 
				$arrRequests,
	]);

$ItemOffersBatchResponse = $ProductPricingApiInstance->getListingOffersBatch($body);

foreach($ItemOffersBatchResponse['responses'] as $Response) // array of responses
{
     $Payload = $Response['body']->payload;
     foreach($Payload->offers as $Offers)
     {
          // You can then access different values like the Buy Box Winner, or the Listing Amount like this:
          $IsBuyBoxWinner = $Offers["is_buy_box_winner"];
          $ListingPrice = $Offers["listing_price"]->amount
     }
}
[/code] 

Hope that helps.
